### DISCUSSION TITLE
Should the active developers virtually meet?

### DISCUSSION THREAD

**Discussion by bpatrik** [bpatrik/pigallery2#893] [General]
I see recently there are more active developers on the project. And my first part of the year got busy and second one wont be lighter either probably. 

I was considering if we should have some virtual sync so the active developers can have some light agreement on the direction of the project. 

I do not make any promises (also I do not expect that from anyone). This is an on-demand hobby project, not a payed job.

I was thinking about meeting in a one big (expect 2-3ppl) zoom/skype whatever room.

Let me know what you think.

Not inclusively to:
@mcdamo @grasdk @mblythe86 etc.

**Comment by mblythe86** [comment]
Sounds good to me.  I'm in the US/Mountain time zone, and tend to be a night owl.  But I'm flexible on time.

**Reply by bpatrik** [reply]
Sounds good, I'm in CET timezone, but I'm used to having meetings with California.

**Reply by grasdk** [reply]
CEST here aswell.

**Reply by kagahd** [reply]
me too, CEST

**Comment by bpatrik** [comment]
Great! Let's do this then.

Can you contact me through a different path to agree on a time? (e.g: linked in, should be easy to find). I prefer not to write my mail down here, because ppl will write me bug reports there :)

**Reply by kagahd** [reply]
I sent you an invite because without a linkedin premium account, one can't message you but send an invite.

**Comment by grasdk** [comment]
Hey guys

Thanks for the talk the other day. It was nice seing you (@bpatrik, @mblythe86, @kagahd). I forgot to ask two things of you. The first one is probably mostly for Patrik:

**Issue handling:**
- When is an issue closed? 
  - On merge of solution?
  - On confirmation from issue-author?
  - ?
- Who closes it? 
  - Issue author?
  - Programmer who solves the issue?
  - Merger (Patrik)?
  - ?

**Development**
- What is your development setup for pigallery2

> I'm asking because I'm not satisfied with mine. 
>- I'm using VS Code and WSL2 under Windows 10 (now 11). 
>- Using mocha test plugin and the VS Code test explorer to run individual or sets of tests.
>- Building and starting pigallery via command line,
>
>- Connecting to WSL with VS code is slow and requires constant window reloads in VS Code to function properly.
>- Running directly under Windows without WSL is faster, but it's very different from the end-environment that pigallery runs on and I don't want to deal with too many issues arising from differences in OS'es.
>- I cannot run any unit tests involving SQLite og MSSQL, either under WSL2 or directly in Windows.
>
> I'm considering a containerized development environment, but before throwing myself into that, I would be happy to hear what you are using :)

**Comment by bpatrik** [comment]
sorry for the late reply, i got super busy lately until like sept. 

**About the issue**: so far I haven't had any strict policy.
Ideally: the reported would verify that the problem is solved and then close the issue. 
I think its important that reported verifies that it is resolved, who closes it not so much anymore. 
Only merging a PR and closing an issue usually misses the verification step.

**development:**
* I use JetBrains WebStorm on win 10
* I tried to set up some lint and code formatting (which partially works)
* I do not run any WSL i did not have any major issue because of that. CI tests run on linux anyway.
* you should be able to SQLite tests, that driver is installed with npm install and works without issue on Win. If you have a MYSQL server on you machine, you can somehow connect to but I forgot how. You can also just disable MySQL tests. There is an env variable to do so.

**Reply by grasdk** [reply]
okay, is there a test runner in WebStorm, or do you run the test by a command? If so, which command. And I'm also interested in the variable to set to enable / disable MySQL.

When I'm using the test-runner in VS Code it gets timeouts on many of the SQLite tests, and the MySQL tests fail, because I don't have a DB running ;)

**Reply by bpatrik** [reply]
I have these set up in webstorm:
![kép](https://github.com/bpatrik/pigallery2/assets/7512182/7a6c3970-9654-432b-a54f-e461d90c2322)

This is how I run sql tests: 
![kép](https://github.com/bpatrik/pigallery2/assets/7512182/1776b8c2-0cc3-44d1-a3ea-35f84956ad2e)

**Reply by bpatrik** [reply]
btw: If I would start today, I would probably use VS code too. So do not feel obliged to switch. :)
